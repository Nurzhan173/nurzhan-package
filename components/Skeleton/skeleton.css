.skeleton-item-wrapper {
  display: flex;
  gap: 10px;
  flex-wrap: wrap;
}

.skeleton-item {
  display: flex;
  flex-direction: column;
  flex-basis: 20%;
  padding: 8px;
  background-color: #f0f0f0;
  border-radius: 8px;
}

.skeleton-header {
  width: 100%;
  height: 166px;
  background-color: #ccc;
  border-radius: 4px;
  margin-bottom: 10px;
  animation: pulse 0.5s ease-in-out infinite alternate;
}

.skeleton-title {
  width: 100%;
  height: 20px;
  background-color: #ccc;
  border-radius: 4px;
  margin-bottom: 10px;
  animation: pulse 0.5s ease-in-out infinite alternate;
}

@keyframes pulse {
  from {
    opacity: 0.6;
  }
  to {
    opacity: 1;
  }
}
