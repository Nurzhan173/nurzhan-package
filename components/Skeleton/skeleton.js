class Skeleton {
  render() {
    const skeleton = document.createElement('div');
    const skeletonItemWrapper = document.createElement('div');
    const skeletonItem = document.createElement('div');
    const skeletonHeader = document.createElement('div');
    const skeletonTitle = document.createElement('div');

    const itemFragment = document.createDocumentFragment();
    const cardFragment = document.createDocumentFragment();

    skeleton.className = 'skeleton';
    skeletonItemWrapper.className = 'skeleton-item-wrapper';
    skeletonItem.className = 'skeleton-item';
    skeletonHeader.className = 'skeleton-header';
    skeletonTitle.className = 'skeleton-title';

    itemFragment.append(skeletonHeader);
    itemFragment.append(skeletonTitle);

    skeletonItem.appendChild(itemFragment)

    for(let i = 0; i < 4; i++){
      cardFragment.append(skeletonItem);
    }

    skeletonItemWrapper.appendChild(cardFragment);

    skeleton.appendChild(skeletonItemWrapper);

    return skeleton;
  }
}

module.exports = Skeleton;
