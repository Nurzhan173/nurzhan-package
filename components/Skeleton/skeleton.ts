import './skeleton.css';

/**
 * @module Skeleton
 */

/**
 * Функция, складывающая два числа.
 *
 * @param {function} onclick - onClick
 * @param {number} b - Второе число.
 * @returns {number} Сумма двух чисел.
 */


class Skeleton {
  render() {
    const skeleton = document.createElement('div');
    const skeletonItemWrapper = document.createElement('div');
    const skeletonItem = document.createElement('div');
    const skeletonHeader = document.createElement('div');
    const skeletonTitle = document.createElement('div');

    const itemFragment = document.createDocumentFragment();
    const cardFragment = document.createDocumentFragment();

    skeleton.className = 'skeleton';
    skeletonItemWrapper.className = 'skeleton-item-wrapper';
    skeletonItem.className = 'skeleton-item';
    skeletonHeader.className = 'skeleton-header';
    skeletonTitle.className = 'skeleton-title';

    itemFragment.append(skeletonHeader);
    itemFragment.append(skeletonTitle);

    skeletonItem.appendChild(itemFragment)

    for(let i = 0; i < 4; i++){
      cardFragment.append(skeletonItem);
    }

    skeletonItemWrapper.appendChild(cardFragment);

    skeleton.appendChild(skeletonItemWrapper);

    return skeleton;
  }
}

export default Skeleton;
