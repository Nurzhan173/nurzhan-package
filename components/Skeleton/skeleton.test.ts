import Skeleton from './skeleton';
import { describe, expect, it } from '@jest/globals'

describe('Skeleton', () => {

  it('should render 4 skeleton items', () => {
    const styles = { color: 'blue' };
    const skeleton = new Skeleton(styles);

    const renderedSkeleton = skeleton.render();

    const skeletonItems = renderedSkeleton.querySelectorAll('.skeleton-item');

    expect(skeletonItems.length).toBe(1);
  });
});
