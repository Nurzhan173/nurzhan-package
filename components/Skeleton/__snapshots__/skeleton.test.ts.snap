// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skeleton should render 4 skeleton items 1`] = `
Skeleton {
  "styles": {
    "color": "blue",
  },
}
`;
